db/migrate/20191119111806_split_disk_type_to_thin_and_format_for_disk.rb

```diff
@@ -0,0 +1,64 @@
+class SplitDiskTypeToThinAndFormatForDisk < ActiveRecord::Migration[5.1]
+  class ExtManagementSystem < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+  end
+
+  class Storage < ActiveRecord::Base
+    include ActiveRecord::IdRegions
+
+    self.inheritance_column = :_type_disabled
+    belongs_to :ext_management_system, foreign_key: "ems_id"
+  end
+
+  class Disk < ActiveRecord::Base
+    include ActiveRecord::IdRegions
+
+    self.inheritance_column = :_type_disabled
+    belongs_to :storage
+    has_one :ext_management_system, through: :storage
```

Comment:
I think the more "correct" association to an EMS would be through vms -> hardwares -> disks.

E.g. we should try to follow https://github.com/ManageIQ/manageiq/blob/master/app/models/ext_management_system.rb#L72